pkg/booter/netbooter.go

```diff
@@ -0,0 +1,82 @@
+package booter
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+)
+
+// NetBooter implements the Booter interface for booting over DHCPv6.
+// See NewNetBooterDHCPv6 for details on the fields.
+type NetBooter struct {
+	Type        string `json:"type"`
+	Method      string `json:"method"`
+	MAC         string `json:"mac"`
+	OverrideURL string `json:"override_url,omitempty"`
+}
+
+// NewNetBooter parses a boot entry config and returns a Booter instance, or an
+// error if any
+func NewNetBooter(config []byte) (Booter, error) {
+	// The configuration format for a NetBooterDHCPv6 entry is a JSON with the
+	// following structure:
+	// {
+	//     "type": "netboot",
+	//     "method": "<method>",
+	//     "mac": "<mac_addr>",
+	//     "override_url": "<url>"
+	// }
+	//
+	// `type` is always set to "netboot".
+	// `method` is one of "dhcpv6", "slaac" or "dhcpv4".
+	// `mac` is the MAC address of the interface to use to boot.
+	// `override_url` is an optional URL used to override the boot file URL used
+	//   to fetch the network boot program. This field becomes mandatory if
```

Comment:
Line indetation

Reply:
just checked - this is intentional indentation to distinguish the various items (method, mac, and override_url)

Reply:
okay